staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/dispatcher.go

```diff
@@ -56,7 +57,20 @@ func newMutatingDispatcher(p *Plugin) func(cm *webhook.ClientManager) generic.Di
 var _ generic.Dispatcher = &mutatingDispatcher{}
 
 func (a *mutatingDispatcher) Dispatch(ctx context.Context, attr *generic.VersionedAttributes, o admission.ObjectInterfaces, relevantHooks []*v1beta1.Webhook) error {
+	reinvokeCtx := attr.GetReinvocationContext()
+	if reinvokeCtx.IsReinvoke() && reinvokeCtx.IsOutputChangedSinceLastWebhookInvocation(attr.Attributes.GetObject()) {
+		// If the object has changed, we know the in-tree plugin re-invocations have mutated the object,
+		// and we need to reinvoke all eligible webhooks.
+		reinvokeCtx.RequireReinvokingPreviouslyInvokedPlugins()
+	}
+	defer func() {
+		reinvokeCtx.SetLastWebhookInvocationOutput(attr.Attributes.GetObject())
```

Comment:
am not a fan of this heavy reinvoke interface and this deep knowledge about the chain logic here. All this logic belongs into the chain IMO. We are leaking chain logic into plugins. Not good.

Reply:
currently to the chain, mutating webhook is a single plugin managing multiple hooks. Therefore hook-level logic is performed inside of the plugin, and the state (context) needs to be carried over to the second pass of the chain

Reply:
Yes, the handler is only aware of plugins, and not able to distinguish between in-tree and webhook plugins, so we don't have the information needed to implement https://github.com/kubernetes/kubernetes/pull/78080#discussion_r286128070 directly in the chain.

Reply:
I'm in favor of making the chain more intelligent eventually, at least being able to distinguish the webhook plugin from built-in plugins, but that's a different topic

Reply:
I've merged in https://github.com/jpbetz/kubernetes/pull/1 from @sttts to slim down the interface.